spec/models/provider_enrichment_spec.rb

```diff
@@ -65,4 +65,22 @@
       end
     end
   end
+
+  describe 'before_create hook' do
+    # Note: provider_code is only here to support c# counterpart, until provide_code is removed from database
+    let(:provider) { build(:provider) }
+    let(:existing_provider_code) { nil }
+    let(:enrichment) { create(:provider_enrichment, provider: provider, provider_code: existing_provider_code) }
+    subject { enrichment }
+
+    context 'when provider_code is blank' do
+      its(:provider_code) { should eq(provider.provider_code) }
+    end
+
+    context 'when provider code is already set' do
+      let(:existing_provider_code) { "XX4" }
+
+      its(:provider_code) { should eq(existing_provider_code) }
```

Comment:
this looks wrong

as 

```ruby 
its(:provider_code) { should eq(provider.provider_code) }
```
would fail

can a enrichment with
a provider that has `provider code`=`xxx`
has a `provider code`=`yyy` that is not the same as the provider

Reply:
This test was more about ensuring that if for whatever reason later the `if provider_code.blank?` gets removed we have some visibility over the potential impact, rather than the _correct_ code being used.